### PR TITLE
Add SelfLink to Kubelet http source.

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

**Comment by vmarmol** [comment]
PTAL @yujuhong

File: pkg/kubelet/config/http.go [hunk 1]
```diff
@@ -206,5 +206,9 @@ func applyDefaults(pod *api.Pod, url string) error {
 	// Always overrides the namespace.
 	pod.Namespace = kubelet.NamespaceDefault
 	glog.V(5).Infof("Using namespace %q for pod %q from URL %s", pod.Namespace, pod.Name, url)
+
+	// TODO(dchen1107): Remove when Kubelet uses pods directly.
```
Comment (yujuhong) [inline pkg/kubelet/config/http.go]: nit: The TODO seems obsolete now that kubelet actually uses pods...

**Comment by yujuhong** [comment]
LGTM. I think we should probably decide how to handle self links better, but that's for future work :)

File: pkg/kubelet/config/http.go [hunk 2]
```diff
@@ -206,5 +206,9 @@ func applyDefaults(pod *api.Pod, url string) error {
...
+	pod.ObjectMeta.SelfLink = fmt.Sprintf("/api/v1beta2/pods/%s?namespace=%s",
```
Comment (dchen1107) [inline pkg/kubelet/config/http.go]: I remembered that I introduced this a while back, why we removed it at the first place?

File: pkg/kubelet/config/http.go (hunk 1 above)
Comment (vmarmol) [inline pkg/kubelet/config/http.go]: Yeah I didn't know how to better express the need to do this better. Removed this TODO and on the other file.

File: pkg/kubelet/config/http.go (hunk 2 above)
Comment (vmarmol) [inline pkg/kubelet/config/http.go]: I tried looking at history but didn't find anything :(

Comment (yujuhong) [inline pkg/kubelet/config/http.go]: I don't think we did. It was there in file.go, but was never(?) in http.go.

**Comment by dchen1107** [comment]
LGTM

**Comment by yujuhong** [comment]
Will merge on green.

**Comment by vmarmol** [comment]
Thanks for the quick review @yujuhong @dchen1107!

**Pull request merged**